### PR TITLE
Generalize node search logic

### PR DESCRIPTION
**Pull request by pontusab**
## Summary
This PR fixes the `node` path issue in current RC `0.64.0-rc.3` https://github.com/react-native-community/releases/issues/214#issuecomment-774486279

Basically extracting the logic to find the correct node path and reusing it in the `generate-specs.sh` file.

## Changelog
[Internal] [Fixed] - Fix node path

## Test Plan
Build the app in Xcode and it should build successfully using with and without nvm.

### CODE REVIEW

**Comment by EdwardDrapkin** [comment]
Why not just run `which node`?

**Comment by ivanmoskalev** [comment]
@EdwardDrapkin This pull request fixes [issues](https://github.com/react-native-community/releases/issues/214#issuecomment-774486279) that occurred when Node was installed with `nvm`.

**Comment by EdwardDrapkin** [comment]
Hard-coding a specific fix for nvm won't solve issues for other more esoteric options, e.g. nvs or graalvm.  Why not just run `which node` at some point when node is available and export that as a variable?  A solution with hardcoded support for three of the dozens of options isn't really generalized and there's got to be a solution that is.

**Comment by ivanmoskalev** [comment]
@EdwardDrapkin The `which node` approach has been tried originally, and it didn't really work. You can read the thread that is linked to this PR, if you like. 

In a nutshell: `generate-specs.sh` and `react-native-xcode.sh` are run as part of an Xcode build step using `/bin/sh`, which doesn't use your `{your_favourite_shell}` profile. That's why you have to call the likes of `$HOME/.nvm/nvm.sh` manually in the script.

It is easily reproducible:
```
$ /bin/sh
sh-3.2$ which node # Nothing.
```

I agree that there should be a better solution for Xcode integrations, and that current scripts feel rather hacky. But I personally don't see any better solution just yet. You'd have to do some form of the lookup dance anyway.

**Comment by grabbou** [comment]
Note: make sure to add `find-node.sh` to `package.json` "files" section.

**Comment by pontusab** [comment]
> Note: make sure to add `find-node.sh` to `package.json` "files" section.

Fixed!

**Comment by pontusab** [comment]
Out in RN 0.64!

**Comment by kelset** [comment]
@pontusab I don't think the PR should be closed, unless I'm mistaken @grabbou manually ported this to the 0.64 branch but this means that main doesn't have the fix yet - reason why we should keep this open and merge it asap :)

**Comment by pontusab** [comment]
Ah gotcha!

**Comment by grit96** [comment]
It would be worth adding `unset npm_config_prefix` to the top of `find-node.sh` to fix [this issue](https://github.com/react-native-community/upgrade-support/issues/138)

File: scripts/find-node.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+set -e
+
```
Comment (grit96) [inline scripts/find-node.sh:8]: Fixes https://github.com/react-native-community/upgrade-support/issues/138#issuecomment-798788120

```suggestion

unset npm_config_prefix

```

**Comment by eugenehp** [comment]
@pontusab 
This  `node-find.sh` pulls old version of node from NVM, even if there's a newer version of node available in the system. In my case via `n`.

**Comment by pontusab** [comment]
@hramos Hey I notice this is now on main added by you, should we close this then?

**Comment by airtonix** [comment]
So why aren't we just doing : 
```
case $SHELL
  zsh) source ~/.zshrc
  bash) source ~/.bashrc
  fish) source ~/.fishrc 
esac
```

or something like that.

**Comment by sota000** [comment]
As far as I can tell, these changes have been merged. Closing the PR. Thanks for the contribution!

**Pull request closed without merging**